packages/syft/src/syft/oblv/model.py

```diff
@@ -0,0 +1,441 @@
+# stdlib
+from datetime import datetime
+import json
+import os
+from signal import SIGTERM
+import subprocess
+import sys
+import time
+from typing import Any
+from typing import List
+
+# third party
+from oblv import OblvClient
+import requests
+
+# relative
+from ..core.node.common.exceptions import OblvEnclaveError
+from ..core.node.common.exceptions import OblvUnAuthorizedError
+from ..util import bcolors
+from .oblv_proxy import check_oblv_proxy_installation_status
+
+
+class DeploymentClient:
+
+    # __attr_allowlist__ = ["deployment_id", "user_key_name", "client", "oblv_client"]
+
+    deployment_id: str
+    user_key_name: str
+    client: List[Any] = []  # List of domain client objects
+    oblv_client: OblvClient = None
+    __conn_string: str
+    __logs: Any
+    __process: Any
+
+    def __init__(
+        self,
+        domain_clients: List[Any],
+        deployment_id: str = None,
+        oblv_client: OblvClient = None,
+        user_key_name: str = "",
+    ):
+        self.deployment_id = deployment_id
+        self.user_key_name = user_key_name
+        self.oblv_client = oblv_client
+        self.client = domain_clients
+        self.__conn_string = ""
+        self.__process = None
+        self.__logs = None
+
+    def set_conn_string(self, URL):
+        self.__conn_string = URL
+
+    def initiate_connection(self, connection_port: int = 3032):
+        if check_oblv_proxy_installation_status() is None:
+            return
+        self.close_connection()  # To close any existing connections
+        public_file_name = os.path.join(
+            os.path.expanduser("~"),
+            ".ssh",
+            self.user_key_name,
+            self.user_key_name + "_public.der",
+        )
+        private_file_name = os.path.join(
+            os.path.expanduser("~"),
+            ".ssh",
+            self.user_key_name,
+            self.user_key_name + "_private.der",
+        )
+        log_file_name = os.path.join(
+            os.path.expanduser("~"),
+            ".oblv_syft_logs",
+            "proxy_logs_" + datetime.now().strftime("%d_%m_%Y_%H_%M_%S") + ".log",
+        )
+        # Creating directory if not exist
+        os.makedirs(os.path.dirname(log_file_name), exist_ok=True)
+        log_file = open(log_file_name, "wb")
+        depl = self.oblv_client.deployment_info(self.deployment_id)
+        if depl.is_deleted:
+            raise Exception(
+                "User cannot connect to this deployment, as it is no longer available."
+            )
+        try:
+            if depl.is_dev_env:
+                process = subprocess.Popen(
```

Comment:
OUGHTTO: I think we could combine line 84 and 110 into one single `Popen` with different arguments -- in the `if...else` we populate the arguments.

Reply:
Placeholder for next task